src/App.tsx

```diff
@@ -1,73 +1,111 @@
-import React from 'react';
-import './App.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import cn from 'classnames';
+import debounce from 'lodash.debounce';
+
+import { Person } from './types/Person';
 import { peopleFromServer } from './data/people';
 
+import './App.scss';
+
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [people, setPeople] = useState(peopleFromServer);
+  const [activePerson, setActivePerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [appliedQuery, setAppliedQuery] = useState<string>('');
+
+  const searchQuery = useCallback(debounce(setAppliedQuery, 300), [
+    setAppliedQuery,
+  ]);
+
+  useMemo(() => {
+    const filteredPeople = peopleFromServer.filter((person: Person) =>
+      person.name.toLowerCase().includes(appliedQuery.toLowerCase()),
+    );
+
+    setPeople(filteredPeople);
+  }, [appliedQuery]);
+
+  const isNotPeople = people.length === 0;
+
+  const onSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    searchQuery(event.target.value);
```

Comment:
The filter should not trigger if only spaces are entered. Add a condition to handle this case.